.github/workflows/native-build.yml

```diff
@@ -39,12 +39,12 @@ jobs:
         include:
           - compiler: clang
             compiler-version: 16
-            asan-flags: "-fsanitize=address -fno-omit-frame-pointer"
+            asan-flags: "-fsanitize=address -fno-omit-frame-pointer -g"
```

Comment:
The more idiomatic (and cleaner) way to achieve this is to specify `build-type: "RelWithDebInfo"` (not sure if the quotes are needed) for those configurations, this expresses the intent more clearly and is more compiler-agnostic.